test/test_prototype_transforms_functional.py

```diff
@@ -1031,3 +1047,46 @@ def _compute_expected(mask, top_, left_, height_, width_, size_):
     expected_mask = _compute_expected(in_mask, top, left, height, width, size)
     output_mask = F.resized_crop_segmentation_mask(in_mask, top, left, height, width, size)
     torch.testing.assert_close(output_mask, expected_mask)
+
+
+def test_correctness_pad_segmentation_mask_on_fixed_input(device):
+    mask = torch.ones((1, 3, 3), dtype=torch.long, device=device)
+
+    out_mask = F.pad_segmentation_mask(mask, padding=[1, 1, 1, 1])
+
+    expected_mask = torch.zeros((1, 5, 5), dtype=torch.long, device=device)
+    expected_mask[:, 1:-1, 1:-1] = 1
+    torch.testing.assert_close(out_mask, expected_mask)
+
+
+@pytest.mark.parametrize("padding", [[1, 2, 3, 4], [1], 1, [1, 2]])
+def test_correctness_pad_segmentation_mask(padding):
+    def _compute_expected_mask():
+        def parse_padding():
+            if isinstance(padding, int):
+                return [padding] * 4
+            if isinstance(padding, list):
+                if len(padding) == 1:
+                    return padding * 4
+                if len(padding) == 2:
+                    return padding * 2  # [left, up, right, down]
+
+            return padding
+
+        h, w = mask.shape[-2], mask.shape[-1]
+        pad_left, pad_up, pad_right, pad_down = parse_padding()
+
+        new_h = h + pad_up + pad_down
+        new_w = w + pad_left + pad_right
+
+        new_shape = (*mask.shape[:-2], new_h, new_w) if len(mask.shape) > 2 else (new_h, new_w)
+        expected_mask = torch.zeros(new_shape, dtype=torch.long)
+        expected_mask[..., pad_up:-pad_down, pad_left:-pad_right] = mask
+
+        return expected_mask
+
+    for mask in make_segmentation_masks():
+        out_mask = F.pad_segmentation_mask(mask, padding, "constant")
+
+        expected_mask = _compute_expected_mask()
+        torch.testing.assert_close(out_mask, expected_mask)
```

Comment:
Thinking of how to test all other padding modes. Maybe we could just check output value from `out_mask` for two lines: one horizontal and one vertical instead of constructing full expected mask.
While checking the lines we still need to route the checks according to the padding mode.
What do you think ?

Reply:
Seems pretty reasonable.

Anyway, the code needed for constructing the two lines IMO still "mimic" (or re-implement) the padding operation.

Reply:
> Thinking of how to test all other padding modes. 

I don't think we need a correctness check for them. Internally, [`torch.nn.functional.pad`](https://pytorch.org/docs/master/generated/torch.nn.functional.pad.html) does the heavy lifting. Thus, if we rely on that giving us the correct behavior there is no need to check if the values of the padding are correct.

Reply:
@pmeier if we do not check correctness on the output, another option is to mock `torch_pad` and ensure that it is called with correct configuration otherwise the code is not covered.